### PR TITLE
CAL-750: add base support for sql-datasets

### PR DESCRIPTION
**Pull request by pcerny**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by jaceksan — APPROVED** [review]

File: docs/content/en/docs/data/data-source/scan_sql.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+linkTitle: "scan_sql"
+weight: 210
+superheading: "catalog_data_source."
+---
+
+
+
+``scan_sql(data_source_id: str, sql_request: ScanSqlRequest)``
+
+Analyze SELECT SQL query in the request. Return description of SQL result-set as list of column names with GoodData data types and list of example data returned by SQL.
```
Comment (Mara3l) [inline docs/content/en/docs/data/data-source/scan_sql.md:12]: Personally I would change "in the request" to "in a given request". I would also consider not repeating the description of the return value.

File: docs/content/en/docs/data/data-source/scan_sql.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{{< parameter p_name="sql_request" p_type="ScanSqlRequest" >}}
+SELECT SQL query to analyze.
+{{< /parameter >}}
+
+{{% /parameters-block %}}
+
+{{% parameters-block title="Returns"%}}
+
+{{< parameter p_type="ScanSqlResponse" >}}
+Description of SQL result-set as list of column names with GoodData data types and list of example data returned by SQL.
```
Comment (Mara3l) [inline docs/content/en/docs/data/data-source/scan_sql.md:29]: I would change returned by SQL to returned by the SELECT query.

**Review by Mara3l — COMMENTED** [review]
It is correct from technical POV, but I would like to change the wording a little bit.

File: docs/content/en/docs/data/data-source/scan_sql.md (hunk 1 above)
Comment (pcerny) [reply docs/content/en/docs/data/data-source/scan_sql.md]: Fixed

File: docs/content/en/docs/data/data-source/scan_sql.md (hunk 2 above)
Comment (pcerny) [reply docs/content/en/docs/data/data-source/scan_sql.md]: Fixed

**Review by Mara3l — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR217

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul/47/ : SUCCESS in 1m 57s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/197/ : SUCCESS in 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/200/ : SUCCESS in 17s

**Pull request merged**